Fix create wager bug

Slenzy2 opened the pull request:
## Description
This PR contains implementations that fixes the bug related to the create wager flow. It specifically fixes the way the form data is being sent to the smart contract it is meant to interact with. Improvements were also made on some utils functions, and few other components related to the create wager flow

## Type of Change
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist
- [x] My code follows the code style of this project.

## Related Issue
- Closes #140 

## Screenshots (if applicable)

Slenzy2 committed:
refactor: streamline wager creation process and enhance contract integration

- Removed unnecessary useEffect hooks and console logs from createWager context and dashboard components.
- Updated wager data validation to replace spaces with hyphens in wager titles.
- Introduced WAGER_ABI in the contract constants and removed the deprecated contracts.ts file.
- Improved error handling in the useCreateWager hook and adjusted U256 conversion logic to align with contract expectations.

Ayoazeez26 commented:
<img width="648" alt="image" src="https://github.com/user-attachments/assets/013f6d9e-0693-489e-95a4-4f8ddb5658ca" />

Update here with the correct username and profile picture of the user

Ayoazeez26 commented:
<img width="711" alt="image" src="https://github.com/user-attachments/assets/70c7a422-5819-4036-88e8-8c5a3e1c0abe" />

Display accurate wallet balance here

Slenzy2 committed:
refactor: update wager schema to use dynamic user balance

- Replaced static user balance with a dynamic value fetched from the wallet context.
- Created a function to generate the wager schema based on the current user balance.
- Simplified type inference for wager form data.

Slenzy2 committed:
refactor: rename context import path and remove unused createWager context

- Updated import paths for CreateWagerProvider and useCreateWagerContext to reflect new directory structure.
- Deleted the obsolete createWager.context.tsx file as it is no longer in use.

Slenzy2 committed:
refactor: enhance avatar handling in BattleDisplay and integrate user profile in WagerSummary

- Updated BattleDisplay component to conditionally render SVG avatars using dangerouslySetInnerHTML for player avatars.
- Modified WagerSummary to fetch and clean user profile data from localStorage, allowing dynamic avatar and username display.
- Improved user experience by ensuring avatars are displayed correctly based on user profile availability.

Slenzy2 commented:
All changes have been made.

Ayoazeez26 reviewed (approved):
Perfect. lgtm
